packages/react-static-tweets/src/twitter.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { createContext, ReactNode, useContext } from 'react'
+import { ConfigInterface } from 'swr'
+
+// TODO: make this more specific
+export type TweetAst = Array<any>
+
+export type TwitterContextValue = {
+  // static tweet ast info
+  tweetAstMap: TweetAstMap
+
+  // SWR config for dynamically fetching tweet ast info
+  swrOptions: ConfigInterface
+}
+
+export type TweetAstMap = {
+  [tweetId: string]: TweetAst
+}
+
+// Saves the tweets returned as props to the page
+const TwitterContext = createContext<TwitterContextValue>({
+  tweetAstMap: {},
+  swrOptions: {
+    fetcher: (id) =>
+      fetch(`https://twitter-search.vercel.app/api/get-tweet-ast/${id}`).then(
+        (r) => r.json()
+      )
+  }
+})
+
+export function useTwitterContext () {
+  return useContext(TwitterContext)
+}
+
+type TwitterContextProviderProps = {
+  value: Partial<TwitterContextValue>
+  children?: ReactNode
+}
+
+// allows partials that override outer providers
+export function TwitterContextProvider ({ value, children }: TwitterContextProviderProps) {
```

Comment:
if you're exporting the Provider, I would expect that you're also exporting the Consumer

Reply:
Agreed. Will do.

Reply:
Done